libs/designsystem/accordion/src/accordion.directive.ts

```diff
@@ -1,8 +1,44 @@
-import { Directive } from '@angular/core';
+import {
+  AfterContentInit,
+  ContentChildren,
+  Directive,
+  Input,
+  OnDestroy,
+  QueryList,
+} from '@angular/core';
+
+import { Subscription } from 'rxjs';
+
+import { AccordionItemComponent } from './accordion-item.component';
 
 @Directive({
   selector: '[kirbyAccordion], kirby-accordion',
   exportAs: 'kirby-accordion',
   standalone: false,
 })
-export class AccordionDirective {}
+export class AccordionDirective implements AfterContentInit, OnDestroy {
+  @Input() headingLevel: number;
+  @ContentChildren(AccordionItemComponent) accordionChildren: QueryList<AccordionItemComponent>;
+
+  private childrenChangesSub?: Subscription;
+
+  ngAfterContentInit(): void {
+    this.setHeadingLevels();
+
+    this.childrenChangesSub = this.accordionChildren.changes.subscribe(() => {
+      this.setHeadingLevels();
+    });
```

Comment:
We should be able to do this in one sweep to both get the first change and subscribe to subsequent changes:
```suggestion
   this.accordionChildren.changes.pipe(startWith(0)).subscribe(() => {
      this.setHeadingLevels();
    });
```